vendor/bat-native-ads/src/bat/ads/internal/creatives/notification_ads/notification_ad_manager.cc

```diff
@@ -101,394 +74,91 @@ NotificationAdManager::MaybeGetForPlacementId(
 
-void NotificationAdManager::PopFront(const bool should_dismiss) {
-  if (!notification_ads_.empty()) {
-    if (should_dismiss) {
-      AdsClientHelper::GetInstance()->CloseNotificationAd(
-          notification_ads_.front().placement_id);
-    }
-    notification_ads_.pop_front();
-    Save();
+    ads_.pop_front();
   }
+
+  AdsClientHelper::GetInstance()->SetListPref(prefs::kNotificationAds,
+                                              NotificationAdsToValue(ads_));
 }
 
 bool NotificationAdManager::Remove(const std::string& placement_id) {
-  DCHECK(is_initialized_);
-
-  auto iter =
-      std::find_if(notification_ads_.cbegin(), notification_ads_.cend(),
-                   [&placement_id](const NotificationAdInfo& notification) {
-                     return notification.placement_id == placement_id;
-                   });
+  DCHECK(!placement_id.empty());
 
-  if (iter == notification_ads_.end()) {
+  const auto iter = std::find_if(ads_.cbegin(), ads_.cend(),
+                                 [&placement_id](const NotificationAdInfo& ad) {
+                                   return ad.placement_id == placement_id;
+                                 });
+  if (iter == ads_.end()) {
     return false;
   }
 
-  notification_ads_.erase(iter);
-  Save();
+  ads_.erase(iter);
 
-  return true;
-}
-
-void NotificationAdManager::RemoveAll() {
-  DCHECK(is_initialized_);
+  AdsClientHelper::GetInstance()->SetListPref(prefs::kNotificationAds,
+                                              NotificationAdsToValue(ads_));
 
-  notification_ads_.clear();
-  Save();
+  return true;
 }
 
-void NotificationAdManager::CloseAndRemoveAll() {
-  DCHECK(is_initialized_);
-
-  for (const auto& notification_ad : notification_ads_) {
-    AdsClientHelper::GetInstance()->CloseNotificationAd(
-        notification_ad.placement_id);
+void NotificationAdManager::CloseAll() {
+  for (const auto& ad : ads_) {
+    AdsClientHelper::GetInstance()->CloseNotificationAd(ad.placement_id);
   }
 
   RemoveAll();
 }
 
 bool NotificationAdManager::Exists(const std::string& placement_id) const {
-  DCHECK(is_initialized_);
+  DCHECK(!placement_id.empty());
 
-  auto iter =
-      std::find_if(notification_ads_.cbegin(), notification_ads_.cend(),
-                   [&placement_id](const NotificationAdInfo& notification) {
-                     return notification.placement_id == placement_id;
-                   });
+  const auto iter = std::find_if(ads_.cbegin(), ads_.cend(),
+                                 [&placement_id](const NotificationAdInfo& ad) {
+                                   return ad.placement_id == placement_id;
+                                 });
 
-  if (iter == notification_ads_.end()) {
+  if (iter == ads_.end()) {
     return false;
   }
 
   return true;
 }
 
-uint64_t NotificationAdManager::Count() const {
-  return notification_ads_.size();
-}
-
-#if BUILDFLAG(IS_ANDROID)
-void NotificationAdManager::RemoveAllAfterReboot() {
```

Comment:
Removing as this code as we will now only cleanup the notification ads pref on browser upgrade